app/models/pawn.rb

```diff
@@ -1,2 +1,42 @@
 class Pawn < Piece
+  def move_legal?(to_row, to_col)
+    return true if single_advance?(to_row, to_col)
+    return true if double_advance?(to_row, to_col)
+    return true if capture_move?(to_row, to_col)
+    return true if en_passant?(to_row, to_col) # need to send signal to capture logic for en passant
+    false
+  end
+
+  def promotion
+  end
+
+  private
+
+  def single_advance?(to_row, to_col)
+    valid_to_row = is_black ? row - 1 : row + 1
+    to_row == valid_to_row && to_col == col
+  end
+
+  def double_advance?(to_row, to_col)
+    valid_to_row = is_black ? 4 : 3
+    to_row == valid_to_row && to_col == col && moves.empty?
+  end
+
+  def capture_move?(to_row, to_col)
+    valid_to_row = is_black ? row - 1 : row + 1
+    to_row == valid_to_row && (to_col - col).abs == 1 && enemy_occupied?(to_row, to_col)
+  end
+
+  def en_passant?(to_row, to_col)
+    valid_to_row = is_black ? row - 1 : row + 1
+    valid_row = is_black ? 4 : 3
+    return false unless moves.length == 1 && row == valid_row
+    return false unless to_row == valid_to_row && (to_col - col).abs == 1
+    enemy_occupied?(row, to_col)
+  end
+
+  def enemy_occupied?(check_row, check_col)
+    piece = Game::Board.new(game.pieces).grid[check_row][check_col]
```

Comment:
This is loading the entire game board from the database in order to retrieve a single piece.  You probably just want the individual piece with `game.pieces.where(row: check_row, col: check_col).first)`.  The captured check could be a part of the query too, or you could even reuse the `are_not_captured` scope:

```ruby
piece = game.pieces.are_not_captured.where(row: check_row, col: check_col).first
```

There's actually already a convenience method `game.piece_at()` defined [here](https://github.com/pawn-stars/chess-app/blob/master/app/models/game.rb#L40) which grabs a piece at a location.  You cannot combine methods with scopes, though, so you could use `game.piece_at(row, col).captured?`, or you could add a `piece_at` scope with arguments to the Piece model and chain it with the `are_not_captured` scope:

```ruby
class Piece < ApplicationRecord
  scope :piece_at, ->(row, col) { where(row: row, col: col) }
end

piece = game.pieces.are_not_captured.piece_at(check_row, check_col)
```